src/Xrefcheck/Verify.hs

```diff
@@ -725,18 +757,26 @@ checkExternalResource Config{..} link
         InvalidUrlException{} -> error "External link URL invalid exception"
         HttpExceptionRequest _ exc -> case exc of
           StatusCodeException resp _
-            | isPermanentRedirectCode code -> Left
-              . PermanentRedirectError link
-              . fmap decodeUtf8
-              . lookup "Location"
-              $ responseHeaders resp
-            | isTemporaryRedirectCode code -> Right ()
-            | isAllowedErrorCode code -> Right ()
+            | Just RedirectRule{..} <- redirectRule link code ncExternalRefRedirects ->
```

Comment:
Right! I will add that check.

> Also, what if we see a redirect but no rule matched?

Before this PR, I explicitly wrote code for fail on permanent redirects and succeed on temporary ones. Now I turned that into the default configuration rules, so in case that no rule matched I think that it will produce an `ExternalHttpResourceUnavailable` verification error.

I'm changing it to pass in that case, but we can continue discussing it (I think that there are other comments related with that).